src/backend/distributed/master/master_create_shards.c

```diff
@@ -167,7 +167,7 @@ CreateShardsWithRoundRobinPolicy(Oid distributedTableId, int32 shardCount,
 	uint64 hashTokenIncrement = HASH_TOKEN_COUNT / shardCount;
 
 	/* don't allow concurrent node list changes that require an exclusive lock */
-	LockRelationOid(DistNodeRelationId(), RowShareLock);
+	LockRelationOid(DistNodeRelationId(), ShareLock);
```

Comment:
Fwiw, RowShare does not conflict with Share. Both conflict with ExclusiveLock (taken by e.g. master_add_node)